api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java

```diff
@@ -506,27 +568,76 @@ public void cleanBillingBuffer_assigning() {
 
   @Test
   public void cleanBillingBuffer_ignoreValidEntries() {
+    DbBillingProjectBufferEntry assigning = new DbBillingProjectBufferEntry();
+    assigning.setStatusEnum(ASSIGNING, this::getCurrentTimestamp);
+    assigning.setCreationTime(getCurrentTimestamp());
+    billingProjectBufferEntryDao.save(assigning);
+
+    CLOCK.setInstant(BEFORE_ASSIGNING_TIMEOUT);
+    billingProjectBufferService.cleanBillingBuffer();
+
+    assertThat(billingProjectBufferEntryDao.findOne(assigning.getId()).getStatusEnum())
+        .isEqualTo(ASSIGNING);
+
     DbBillingProjectBufferEntry creating = new DbBillingProjectBufferEntry();
     creating.setStatusEnum(CREATING, this::getCurrentTimestamp);
     creating.setCreationTime(getCurrentTimestamp());
     billingProjectBufferEntryDao.save(creating);
 
-    CLOCK.setInstant(NOW.plus(59, ChronoUnit.MINUTES));
+    CLOCK.setInstant(BEFORE_CREATING_TIMEOUT);
     billingProjectBufferService.cleanBillingBuffer();
 
     assertThat(billingProjectBufferEntryDao.findOne(creating.getId()).getStatusEnum())
         .isEqualTo(CREATING);
+  }
 
-    DbBillingProjectBufferEntry assigning = new DbBillingProjectBufferEntry();
-    assigning.setStatusEnum(ASSIGNING, this::getCurrentTimestamp);
-    assigning.setCreationTime(getCurrentTimestamp());
-    billingProjectBufferEntryDao.save(assigning);
+  @Test
+  public void cleanBillingBuffer_multipleCreates() {
+    billingProjectBufferService.bufferBillingProjects();
+
+    CLOCK.setInstant(NOW.plus(30, ChronoUnit.MINUTES));
+    billingProjectBufferService.bufferBillingProjects();
+    billingProjectBufferService.bufferBillingProjects();
+
+    final ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+    verify(fireCloudService, times(3)).createAllOfUsBillingProject(captor.capture());
+    final List<String> capturedProjectNames = captor.getAllValues();
+    doReturn(new BillingProjectStatus().creationStatus(CreationStatusEnum.CREATING))
+        .when(fireCloudService)
+        .getBillingProjectStatus(capturedProjectNames.get(0));
+    doReturn(new BillingProjectStatus().creationStatus(CreationStatusEnum.CREATING))
+        .when(fireCloudService)
+        .getBillingProjectStatus(capturedProjectNames.get(1));
+    doReturn(new BillingProjectStatus().creationStatus(CreationStatusEnum.ERROR))
+        .when(fireCloudService)
+        .getBillingProjectStatus(capturedProjectNames.get(2));
+
+    Supplier<List<BillingProjectBufferStatus>> statusSupplier =
```

Comment:
Since this is a fairly new pattern, a little comment here would be helpful I think.

Reply:
Done